src/overmap.cpp

```diff
@@ -1348,139 +1385,95 @@ void overmap::draw(WINDOW *w, WINDOW *wbar, const tripoint &center,
         }
     }
 
-    for (int i = 0; i < om_map_width; i++) {
-        for (int j = 0; j < om_map_height; j++) {
-            const int omx = cursx + i - (om_map_width / 2);
-            const int omy = cursy + j - (om_map_height / 2);
+    // A small LRU cache: most oter_id's occur in clumps like forests of swamps.
```

Comment:
Yes, all of the changes were made piecemeal one by one with measurable results: measure, change, measure. This was one of the bigger improvements if memory serves -- on the order of a 10~15% relative (not global) improvement, but exact numbers I no longer have. In the future perhaps each individual small optimization should be commited separately with a rough estimate of how much it affects performance.